patches/tModLoader/Terraria/NPC.cs.patch

```diff
@@ -774,12 +774,25 @@
  		}
  		else if (type == 607) {
  			noGravity = true;
-@@ -10793,13 +_,18 @@
+@@ -10792,14 +_,31 @@
+ 			DeathSound = SoundID.NPCDeath6;
  			catchItem = 2121;
  		}
- 
-+		NPCLoader.SetDefaults(this);
++		// TML: Moved before NPCLoader.SetDefaults to allow changing. DebuffImmunitySets broken into separate sets for easier manipulation.  
++		if (NPCID.Sets.ImmuneToAllDebuffsByDefault[type])
++			Array.Fill(buffImmune, true);
```

Comment:
Do you need to fill the array regardless? Where does it get cleared?
`Array.Fill(buffImmune, NPCID.Sets.ImmuneToAllDebuffsByDefault[type]);`

Reply:
It gets filled with false at the start of NPC.SetDefaults, so we don't need to add code for that.